frontend/src/store.ts

```diff
@@ -3408,13 +3411,15 @@ export function createStore(web3: Web3) {
         const { CryptoBlades, SkillToken, Blacksmith } = state.contracts();
         if(!CryptoBlades || !Blacksmith || !state.defaultAccount) return;
 
-        await approveFeeFromAnyContractSimple(
+        await approveFeeWalletOrRewards(
+          CryptoBlades,
```

Comment:
Duplicates on a few of these calls?

Reply:
As discussed in Discord, this is the preexisting usage where one of the params is the rewards holder (CryptoBlades) and the other is the fee spending contract (not always CryptoBlades).

A future refactor could maybe make this simpler/smaller for the "they match" case, but that doesn't work for spenders such as Blacksmith, BurningManager, etc.